pbft/pbft.cpp

```diff
@@ -352,21 +352,35 @@ pbft::forward_request_to_primary(const bzn_envelope& request_env)
     this->failure_detector->request_seen(req_hash);
 }
 
+
 void
-pbft::maybe_record_request(const pbft_msg& msg, const std::shared_ptr<pbft_operation>& op)
+pbft::maybe_record_request(const pbft_msg& msg, const bzn_envelope& env, const std::shared_ptr<pbft_operation>& op)
 {
-    if (msg.has_request() && !op->has_request())
+    if (!op->has_request())
     {
-        if (this->crypto->hash(msg.request()) != msg.request_hash())
+        if (msg.request_type() == PBFT_REQUEST_PAYLOAD)
         {
-            LOG(info) << "Not recording request because its hash does not match";
-            return;
+            assert(env.piggybacked_requests_size()); // we must have at least 1 request
+            if (this->crypto->hash(env.piggybacked_requests(0)) != msg.request_hash())
```

Comment:
It may be simpler for this check to be outside the scope of this method - since the caller likely already identified the pbft_operation it passed to us by way of matching the hashes

Reply:
Reviewed with Isabel.